packages/dd-trace/src/plugins/util/web.js

```diff
@@ -197,8 +218,13 @@ const web = {
   },
 
   // Extract the parent span from the headers and start a new span as its child
-  startChildSpan (tracer, name, headers) {
-    const childOf = tracer.scope().active() || tracer.extract(FORMAT_HTTP_HEADERS, headers)
+  startChildSpan (tracer, name, headers, store) {
+    let childOf
+    if (store) {
+      childOf = store ? store.span : store || tracer.extract(FORMAT_HTTP_HEADERS, headers)
+    } else {
+      childOf = tracer.scope().active() || tracer.extract(FORMAT_HTTP_HEADERS, headers)
```

Comment:
Why is the scope manager needed? The scope manager relies on the store, so if there is no store there cannot be an active span.